libraries/cmake/source/ssdeep-cpp/CMakeLists.txt

```diff
@@ -7,35 +7,68 @@
 function(ssdeepCppMain)
   set(library_root "${CMAKE_CURRENT_SOURCE_DIR}/src")
 
-  add_library(thirdparty_ssdeep-cpp
+  add_library(thirdparty_ssdeep-cpp INTERFACE)
+
+  add_library(thirdparty_ssdeep-cpp_cpp
     "${library_root}/match.cpp"
     "${library_root}/engine.cpp"
     "${library_root}/filedata.cpp"
     "${library_root}/dig.cpp"
     "${library_root}/cycles.cpp"
     "${library_root}/helpers.cpp"
     "${library_root}/ui.cpp"
+  )
+
+  add_library(thirdparty_ssdeep-cpp_c
     "${library_root}/fuzzy.c"
     "${library_root}/find-file-size.c"
   )
 
-  target_compile_definitions(thirdparty_ssdeep-cpp PRIVATE
+  target_compile_definitions(thirdparty_ssdeep-cpp_cpp PRIVATE
+    HAVE_CONFIG_H
+  )
+
+  target_compile_definitions(thirdparty_ssdeep-cpp_c PRIVATE
     HAVE_CONFIG_H
   )
 
-  target_link_libraries(thirdparty_ssdeep-cpp PRIVATE
+  target_link_libraries(thirdparty_ssdeep-cpp_cpp PRIVATE
     thirdparty_cxx_settings
     thirdparty_c_settings
   )
 
-  target_include_directories(thirdparty_ssdeep-cpp PRIVATE
-    "${CMAKE_CURRENT_SOURCE_DIR}/config"
-    "${library_root}"
+  target_link_libraries(thirdparty_ssdeep-cpp_c PRIVATE
+    thirdparty_c_settings
   )
 
+  if(DEFINED PLATFORM_LINUX)
+    target_include_directories(thirdparty_ssdeep-cpp_cpp PRIVATE
+      "${CMAKE_CURRENT_SOURCE_DIR}/config/linux"
+      "${library_root}"
```

Comment:
Nitpick: maybe dedup "${library_root}" for both macOS and Linux with:
```
target_include_directories(thirdparty_ssdeep-cpp_cpp PRIVATE
  "${library_root}"
)
target_include_directories(thirdparty_ssdeep-cpp_c PRIVATE
  "${library_root}"
)
```
outside of the if statements, since it's common for both?